csi/controller_server.go

```diff
@@ -646,6 +646,14 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 	}
 
 	_, volumeName, backupName := decodeSnapshotID(req.SnapshotId)
+	snapshot, err := cs.apiClient.Snapshot.ById(backupName)
```

Comment:
1. I think it should be `snapshot, err := cs.apiClient.Snapshot.ById(req.SnapshotId)`
2. There must be an error returned from Longhorn API layer if the snapshot doesn't exist:
```
func (m *VolumeManager) GetSnapshot(snapshotName, volumeName string) (*types.Snapshot, error) {
	if volumeName == "" || snapshotName == "" {
		return nil, fmt.Errorf("volume and snapshot name required")
	}
	engine, err := m.GetEngineClient(volumeName)
	if err != nil {
		return nil, err
	}
	snapshot, err := engine.SnapshotGet(snapshotName)
	if err != nil {
		return nil, err
	}
	if snapshot == nil {
		return nil, fmt.Errorf("cannot find snapshot '%s' for volume '%s'", snapshotName, volumeName)
	}
	return snapshot, nil
}
```
How does `if snapshot == nil` works?

Reply:
Sounds good to me, fixed.